src/test/test_controller.c

```diff
@@ -1689,6 +1691,187 @@ test_download_status_bridge(void *arg)
   return;
 }
 
+/** Mock cached consensus */
+static cached_dir_t *mock_ns_consensus_cache;
+static cached_dir_t *mock_microdesc_consensus_cache;
+
+int
+mock_we_want_to_fetch_flavor(const or_options_t *options,
+                             int flavor);
+
+int
+mock_we_want_to_fetch_flavor(const or_options_t *options,
+                             int flavor) {
+  if (options->FetchUselessDescriptors ||
+      flavor == FLAV_NS || flavor == FLAV_MICRODESC) {
+    return 1;
+  }
+  return 0;
+}
+
+/**  Mock the function that retrieves consensus from
+ * cache. These use a global variable so that they can
+ * be cleared from within the test. The actual code
+ * retains the pointer to the consensus data, but
+ * we are doing this here, to prevent memory leaks
+ * from within the tests */
+static cached_dir_t *
+mock_dirserv_get_consensus(const char *flavor_name);
+
+static cached_dir_t *
+mock_dirserv_get_consensus(const char *flavor_name)
+{
+  if (!strcmp(flavor_name, "ns")) {
+    mock_ns_consensus_cache = tor_malloc_zero(sizeof(cached_dir_t));
+    mock_ns_consensus_cache->dir = tor_strdup("mock_ns_consensus");
+    return mock_ns_consensus_cache;
+  } else {
+    mock_microdesc_consensus_cache = tor_malloc_zero(sizeof(cached_dir_t));
+    mock_microdesc_consensus_cache->dir = tor_strdup(
+                                            "mock_microdesc_consensus");
+    return mock_microdesc_consensus_cache;
+  }
+}
+
+/** Mock for the function that returns the name of the
+ * file containing the consensuses. */
+char *
+mock_networkstatus_get_cache_fname(int flav,
+                                   const char *flavor_name,
+                                   int unverified_consensus);
+
+char *
+mock_networkstatus_get_cache_fname(int flav,
+                                   const char *flavor_name,
+                                   int unverified_consensus)
+{
+  /** A function needs to do something with
```

Comment:
You don't need to do useless stuff with variables. Instead just do `(void) flav` if you have no need for the argument.

Reply:
Thanks, I didn't know I could do that.